sklearn/preprocessing/tests/test_polynomial.py

```diff
@@ -490,6 +491,56 @@ def test_spline_transformer_n_features_out(
     assert splt.transform(X).shape[1] == splt.n_features_out_
 
 
+@pytest.mark.parametrize("knots", ["uniform", "quantile"])
+@pytest.mark.parametrize(
+    "extrapolation", ["error", "constant", "linear", "continue", "periodic"]
+)
+@pytest.mark.parametrize("sparse_output", [False, True])
+def test_spline_transformer_handles_missing_values(knots, extrapolation, sparse_output):
+    """Test that SplineTransformer handles missing values correctly."""
+    X_nan = [[1, 1], [2, 2], [3, 3], [np.nan, 4], [4, 4]]
+    X = [[1, 1], [2, 2], [3, 3], [4, 4]]
+
+    # only for development, will later be removed
+    # X_nan = [[1], [2], [3], [np.nan], [4]]
+    # X = [[1], [2], [3], [4]]
+
+    # check correct error message for handle_missing="error"
+    msg = (
+        "'X' contains np.nan values, which is conflicting with handle_missing='error'."
+    )
+    with raises(ValueError, match=msg):
+        spline = SplineTransformer(
+            degree=2,
+            n_knots=3,
+            knots=knots,
+            handle_missing="error",
+            extrapolation=extrapolation,
+        )
+        spline.fit_transform(X_nan)
+
+    # check correct results for handle_missing="indicator", remark: check only
+    # for knots="uniform", since for "quantile" the metrics will be calculated
+    # differently with nan present and a different result is thus expected
+    spline = SplineTransformer(
+        degree=2,
+        n_knots=3,
+        knots="uniform",
+        handle_missing="indicator",
+        extrapolation=extrapolation,
+        sparse_output=sparse_output,
+    )
+
+    X_nan = np.array(X_nan)
+    mask = _get_mask(X_nan, np.nan)
+
+    X_transform = spline.fit_transform(X)
+    X_nan_transform = spline.fit_transform(X_nan)
+    X_nan_transform_without_extra = X_nan_transform[~mask[:, 0], :-1]
+
+    assert np.array_equal(X_transform, X_nan_transform_without_extra)
```

Comment:
Actually this assertion is not guaranteed to hold when nan values could mask the largest or smallest values in `X` when `knots="uniform"`. Maybe the comment at line 524 could be expanded to make that explicit.

Reply:
I have added a comment for that, but actually this is another example to the fact that making splines of zeros for nan values, changes the transformer's stats significantly. I am confused about whether this is desired or should be avoided.